imagescan/controller.go

```diff
@@ -258,12 +258,20 @@ func (s *Controller) scanImage(ctx context.Context, img *image) (rerr error) {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
-	mode := s.getScanMode(img)
+	// If mode is empty it will be determined automatically based on container runtime inside scanner.go
```

Comment:
Would be nice to have separate method. Since this logic more complex it could also have some separate tests. 
```
func (s *Controller) findBestNode(mode string, img *image) (string, error) {

}
```

Reply:
most of it is already covered by old test and mode is codependent on this node resolution. But I will move it to separate function and add missing tests